src/js/components/ChannelNav/ChannelNav.js

```diff
@@ -0,0 +1,136 @@
+import '../../../scss/components/ChannelNav/ChannelNav.scss';
+import { GlobalContext } from '../../contexts/GlobalContextWrapper';
+import useDiscordData from '../../hooks/useDiscordData';
+import ChannelNavButton, { ChannelNavParent } from '../ChannelNavButton/ChannelNavButton';
+import UserControlPanel from '../UserControlPanel/UserControlPanel';
+
+const ChannelNav = () => {
+  const history = useHistory();
+  const { guild } = useParams();
+
+  const { Config } = useContext(GlobalContext);
+  const { discordAPIResources: { userGuilds, pending } } = Config.get(['discordAPIResources']);
+
+  let guildName = '';
+  let guildId = pending;
+
+  const guildData = useDiscordData(userGuilds);
+
+  if (guildData) {
+    // Checks that the URL param guild id matches the discordAPI user guild id
+    const findGuild = guildData.find((e) => e.id === guild);
+
+    if (findGuild) {
+      if (guild !== findGuild.id) {
+        history.push({
+          pathname: '/error',
+          state: {
+            error: 'Guild ID does not match user guilds',
+          },
+        });
+      }
+
+      const { id, name } = findGuild;
+
+      guildName = name;
+      guildId = id;
+    }
+  }
+
+  return (
+    <div className="nav-column nav-column-channels">
+      <div className="nav-group nav-group-top">
+        <button type="button" className="btn btn-top">
+          <h1>{guildName}</h1>
+          <div className="svg svg-arrow" />
+        </button>
+      </div>
+      <div className="nav-group nav-group-channels">
+        <ChannelNavButtons guildId={guildId} />
+      </div>
+      <div className="nav-group nav-group-foot">
+        <UserControlPanel />
+      </div>
+    </div>
+  );
+};
+
+const ChannelNavButtons = ({ guildId }) => {
+  const { Config } = useContext(GlobalContext);
+  const { discordAPIResources } = Config.get(['discordAPIResources']);
+  const { guilds, channels, parentChannel } = discordAPIResources;
+  const [hideChildrenId, setHideChildrenId] = useState([]);
+
+  const channelsResource = [guilds, guildId, channels].join('/');
+  const guildChannels = useDiscordData(channelsResource);
+
+  const orderChannels = [];
+
+  if (guildChannels) {
+    const { guild_id: guildTestId } = guildChannels[0];
+
+    // Checks the channels received from the Discord API belongs to the current guild in view
+    if (guildTestId !== guildId) {
+      return <></>;
+    }
+
+    // Filters out "parents" and "channels" for easier sorting
+    const parents = guildChannels.filter((e) => e.type === parentChannel);
+    const filteredChannels = guildChannels.filter((e) => e.type !== parentChannel);
+
+    // Sorts channels so that then come after their parent
+    parents.forEach((parent) => {
+      orderChannels.push(parent);
+
+      filteredChannels.forEach((e) => {
+        const { parent_id: parentId } = e;
+
+        if (parentId === parent.id) {
+          orderChannels.push(e);
+        }
+      });
+    });
+  }
```

Comment:
Instead of this boilerplate, could you have done the "filter" during the render? i.e, in `guildChannels.map()` if the channel is a parent, then filter its children and render them as well?

The reason I ask is, when there's this much nesting just for housekeeping, I start to wonder if there's a more streamlined way to do it. Maybe something you could try out next time?